lib/attrsets.nix

```diff
@@ -534,23 +534,42 @@ rec {
         in mapAttrs g;
     in recurse [] set;
 
-
   /* Generate an attribute set by mapping a function over a list of
      attribute names.
 
      Example:
-       genAttrs [ "foo" "bar" ] (name: "x_" + name)
-       => { foo = "x_foo"; bar = "x_bar"; }
+       mapListToAttrs [ "foo" "bar" ] (name: "x_" + name)
+       => { bar = "x_bar"; foo = "x_foo"; }
 
      Type:
-       genAttrs :: [ String ] -> (String -> Any) -> AttrSet
+       mapListToAttrs :: [String] -> (String -> Any) -> AttrSet
   */
-  genAttrs =
-    # Names of values in the resulting attribute set.
-    names:
-    # A function, given the name of the attribute, returns the attribute's value.
+  mapListToAttrs =
+    # A function, given a string, that produces a corresponding value
+    f:
+    # The list of strings to map
+    list:
+    listToAttrs (map (n: nameValuePair n (f n)) list);
+
+  # DEPRECATED
+  genAttrs = lib.warn
+    "lib.genAttrs is deprecated, use lib.mapListToAttrs instead" (flip mapListToAttrs);
```

Comment:
This is an old, widespread function.
Deprecating `genAttrs` now would cause third-party library maintainers to migrate and therefore break with previous nixpkgs versions, among which is the 22.11 installer.
I doubt that it should be deprecated at all because it is widespread, and because having the values and the variable name close together is nice.

Reply:
It might indeed be good to deprecate it at a later time, such as with a release, and leave it as an action item commented as such for later. 

Not deprecating _ever_ because it's used doesn't make much sense to me though, as the name isn't indicative of its purpose in typical functional programming terms, and the parameter ordering as-is honestly doesn't make sense -- it's not consistent with the rest of the `lib.attrsets` functions, and afaik, it's rather standard to have the applied data after any function parameters for higher-order functions in reflection of a typed lambda calculus

Reply:
Removed warning in 6f0562b. Leaving comment unresolved for now.